source/MaterialXRender/Mesh.h

```diff
@@ -275,39 +275,25 @@ class MX_RENDER_API Mesh
 
     /// Get a mesh stream by name
     /// @param name Name of stream
+    /// @param allowFallback If true will attempt to return fallback stream if it does not exist. Default is false.
     /// @return Reference to a mesh stream if found
-    MeshStreamPtr getStream(const string& name) const
-    {
-        for (const auto& stream : _streams)
-        {
-            if (stream->getName() == name)
-            {
-                return stream;
-            }
-        }
-        return MeshStreamPtr();
-    }
+    MeshStreamPtr getStream(const string& name, bool allowFallback = false);
 
     /// Get a mesh stream by type and index
     /// @param type Type of stream
     /// @param index Index of stream
+    /// @param allowFallback If true will attempt to return a fallback stream if it does not exist. Default is false.
     /// @return Reference to a mesh stream if found
-    MeshStreamPtr getStream(const string& type, unsigned int index) const
-    {
-        for (const auto& stream : _streams)
-        {
-            if (stream->getType() == type &&
-                stream->getIndex() == index)
-            {
-                return stream;
-            }
-        }
-        return MeshStreamPtr();
-    }
+    MeshStreamPtr getStream(const string& type, unsigned int index, bool allowFallback = false);
 
     /// Add a mesh stream
     void addStream(MeshStreamPtr stream)
     {
+        for (MeshStreamPtr s : _streams)
```

Comment:
This seems like a surprising behavior for `Mesh::addStream`, where it will silently skip the requested addition when a stream by the same name already exists.  In this situation, the caller has no way to know that the mesh stream that they have constructed is not actually referenced or needed by the mesh.

Would it instead be better to thrown an exception in this situation, so that the logic bug leading to this situation can be addressed?